### PR TITLE
Only use first field maps for shared statics

### PR DESCRIPTION
**Pull request by SingleAccretion**
This change rounds out the story of redundant maps for statics in VN by encoding the shared-ness information in the lower two bits of the handle stored inside the `FieldSeqNode` and using that information in `IsFieldAddr`.

Addresses two `TODO-VNTypes`. Saves `~0.12%` in memory consumption when R2R-ing CoreLib.

No diffs are expected.

Part of #58312.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
Zero diffs recorded: https://dev.azure.com/dnceng/public/_build/results?buildId=1667122&view=ms.vss-build-web.run-extensions-tab.

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -17212,18 +17204,18 @@ void GenTree::ParseArrayAddress(
...
+        if (!FieldSeqStore::IsPseudoField(fldSeqIter->GetFieldHandleValue()))
         {
             if (*pFldSeq == nullptr)
             {
                 *pFldSeq = fldSeqIter;
             }
             CORINFO_CLASS_HANDLE fldCls = nullptr;
-            noway_assert(fldSeqIter->m_fieldHnd != nullptr);
-            CorInfoType cit = comp->info.compCompHnd->getFieldType(fldSeqIter->m_fieldHnd, &fldCls);
+            noway_assert(fldSeqIter->GetNext() != nullptr);
```
Comment (jakobbotsch) [inline src/coreclr/jit/gentree.cpp:17214]: Was this assert intentionally changed?

Comment (SingleAccretion) [reply src/coreclr/jit/gentree.cpp:17214]: No, thank you for noticing it!

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/gentree.cpp [hunk 2]
```diff
@@ -17212,18 +17204,18 @@ void GenTree::ParseArrayAddress(
...
+        if (!FieldSeqStore::IsPseudoField(fldSeqIter->GetFieldHandleValue()))
         {
             if (*pFldSeq == nullptr)
             {
                 *pFldSeq = fldSeqIter;
             }
             CORINFO_CLASS_HANDLE fldCls = nullptr;
-            noway_assert(fldSeqIter->m_fieldHnd != nullptr);
-            CorInfoType cit = comp->info.compCompHnd->getFieldType(fldSeqIter->m_fieldHnd, &fldCls);
+            noway_assert(fldSeqIter->GetFieldHandle() != NO_CLASS_HANDLE);
```
Comment (SingleAccretion) [inline src/coreclr/jit/gentree.cpp:17214]:
```suggestion
            noway_assert(fldSeqIter->GetFieldHandle() != NO_FIELD_HANDLE);
```

**Pull request merged**